### PR TITLE
Linting fixes

### PR DESCRIPTION
**Pull request by xinhaoz**
## Notion ticket link

[Misc. frontend fixes](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d?p=14667708e5b14f16a81d935f5e4c0e6e)

## Implementation description
* Add plugins to eslint config
* Change prettier errors to 'warn' instead of 'error'
* Define 'any' return types
* Add missing return type to functions

## Steps to test
1. Run app, ensure app runs as normal (no new features added)

## What should reviewers focus on?
* frontend/.eslint
* Converted any types

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/APIClients/BaseAPIClient.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ baseAPIClient.interceptors.request.use(async (config: AxiosRequestConfig) => {
     authHeaderParts.length >= 2 &&
     authHeaderParts[0].toLowerCase() === "bearer"
   ) {
+    // TODO: define expected JWT return type
     const decodedToken: any = jwt.decode(authHeaderParts[1]);
```
Comment (xinhaoz) [inline frontend/src/APIClients/BaseAPIClient.ts:37]: @sherryhli @chenrichard10  @zm2he 
Anyone know what is supposed to be the format of the object returned? Changing this from any to the expected object.

File: frontend/src/index.tsx [hunk 2]
```diff
@@ -38,12 +39,13 @@ const link = createHttpLink({
...
-    "accessToken",
-  );
+  let token: string | null = getLocalStorageObjProperty<
+    NonNullable<AuthenticatedUser>,
+    string
+  >(AUTHENTICATED_USER_KEY, "accessToken");
 
   if (token) {
+    // TODO: define expected JWT return type
     const decodedToken: any = jwt.decode(token);
```
Comment (xinhaoz) [inline frontend/src/index.tsx:49]: @sherryhli @chenrichard10 @zm2he 
Same here, (if it's diff from the other one).

File: frontend/src/APIClients/BaseAPIClient.ts (hunk 1 above)
Comment (sherryhli) [reply frontend/src/APIClients/BaseAPIClient.ts:37]: I'll look into it!

Comment (sherryhli) [reply frontend/src/APIClients/BaseAPIClient.ts:37]: Should be `string | { [key: string]: any; } | null` but typing it as such would result in an error when accessing the `exp` property below. Any way to work around this?

While you're here could you add a line to skip the `exp` check if `decodedToken` is a string?

File: frontend/src/index.tsx (hunk 2 above)
Comment (sherryhli) [reply frontend/src/index.tsx:49]: It's the same as above

File: frontend/src/index.tsx [hunk 3]
```diff
@@ -38,18 +39,19 @@ const link = createHttpLink({
...
   if (token) {
-    const decodedToken: any = jwt.decode(token);
+    const decodedToken = jwt.decode(token) as DecodedJWT;
 
     // refresh if decodedToken has expired
     if (
       decodedToken &&
-      decodedToken.exp > Math.round(new Date().getTime() / 1000)
+      (typeof decodedToken === "string" ||
+        decodedToken.exp > Math.round(new Date().getTime() / 1000))
```
Comment (sherryhli) [inline frontend/src/index.tsx:54]: This is totally my bad but can you please change this to `<=`? I forgot to revert after testing 🤦

**Review by sherryhli — APPROVED** [review]
Looks good! There's a small issue unrelated to your changes but if you could include the fix here that would be great

File: frontend/src/index.tsx (hunk 3 above)
Comment (sherryhli) [inline frontend/src/index.tsx:54]:
```suggestion
        decodedToken.exp <= Math.round(new Date().getTime() / 1000))
```

Feel free to just commit this

**Review by chenrichard10 — APPROVED** [review]

**Pull request merged**